src/main/java/com/direwolf20/buildinggadgets/items/CopyPasteTool.java

```diff
@@ -43,10 +43,12 @@
 
 import java.util.*;
 
+import javax.annotation.Nullable;
+
 import static com.direwolf20.buildinggadgets.tools.GadgetUtils.useEnergy;
 import static com.direwolf20.buildinggadgets.tools.GadgetUtils.withSuffix;
 
-public class CopyPasteTool extends GenericGadget {
+public class CopyPasteTool extends GenericGadget implements ITemplate {
 
     public enum toolModes {
```

Comment:
ToolMode (First Character Capitalized and then Singular Enum as by Conventions)? (If I remember correctly, then this would need to be done with the Building Gadget too)